### PR TITLE
Remove strange class limitation for `important`

### PR DESCRIPTION
**Pull request by LoganDark**
I don't see why this has to be here considering it's just a CSS selector.

It really is a strange thing.

### CODE REVIEW

**Comment by jreviews** [comment]
This is related to #1738 which was closed. One particular use case is where you don't have complete control over the page, but you do over sections of the page. Since you can't have duplicate `ids` in HTML, this doesn't help. But it does help to be able to add specificity using a parent class selector.

**Comment by adamwathan** [comment]
This is by design due to the bug mentioned here:

https://github.com/tailwindcss/tailwindcss/issues/1445

Have you checked to see what actually happens with your PR if you use a class for important?

Here's an example of the output when you set `important` to `.example`:

```css
.hover\:example:hover .hover\:bg-white:hover {
  background-color: #fff;
}

.hover\:example:hover .hover\:bg-gray-100:hover {
  background-color: #f7fafc;
}
```

TLDR is it doesn't work, so instead of having people wonder why they are seeing broken CSS generated, we explicitly throw an error. Using a class doesn't actually meaningfully increase the specificity either, so it's not really appropriate to use a class for this option. It's not meant to be used as a general scoping mechanism. We may introduce something like that in the future but not currently in development.

**Comment by jreviews** [comment]
Thanks Adam. I get it. Maybe theres another way to do this? For example, I can wrap:

```
.some-class {
   @tailwind utilities;
}
```

And that seems to work, except for the classes inside media queries. Is there a way to do something similar that includes this inside the media queries?

**Comment by adamwathan** [comment]
Should work as long as you are using `postcss-nested` or similar 👍

**Comment by jreviews** [comment]
Perfect! Thanks so much for Tailwind, and for taking the time to help with this.

**Comment by LoganDark** [comment]
> instead of having people wonder why they are seeing broken CSS generated, we explicitly throw an error.

You should fix the broken CSS generation because that is a bug, but ok. :(

**Comment by adamwathan** [comment]
You can fix the bug too pal 😉 it’s a hard problem to fix properly and this is better than nothing for now. Again it’s also not what this feature is for.

**Comment by LoganDark** [comment]
> You can fix the bug too pal 😉

I was trying to make this change for someone else who was running into the issue of only being able to use an ID for one element. They wanted to use Tailwind in an extension when injecting things into the page, but wanted to be able to have it in more than one spot.

I am not spending a bunch of time setting up Tailwind locally just to fix some broken `important`. It would take me a while to figure out how the code even works, and I'm currently working on something else so it would also take my time away from that.

If you're not willing to fix the bug that causes this problem because it's "not what this feature is for", maybe you should reconsider what this feature is for. Because sometimes people want to scope Tailwind to something other than an ID, and this limitation is just bizarre for those people.

However, I'm not asking you for anything and there's no reason to tell me to do it myself. You already closed this PR because you're not interested in this, and quite frankly I'm not really interested either. I was trying to help someone out by making a quick PR (if you check the commits, they were made from the GitHub website) but clearly that isn't good enough.

I'm not discussing this further. I tried and failed, and I am not wasting any more time on this trivial issue.
